### PR TITLE
Origin/feature/7648 family/7812 refactor certificates screen

### PR DESCRIPTION
**Pull request by pascal-brause**
## Description
Rearranges the certificates overview screen to adopt the new layout.
1. We have now a "add certificate" cell like in the checkin overview
2. We have now an empty state like in the checkin overview
3. we show only 3 types of cells: the add cell, the certificate request cell and the healthCertifiedPerson cell, which is currently in the old state but for this, there is separate subtask ticket.
4. When the system permission for the camera is missing, we show the specialized cell like in the checkin overview.
5. cleanup of old obsolete code
6. moving reused cell AddButtonAsTableViewCell and its model protocol to more generic place in the project folder, therefore it can now be reused (only use the cell, create custom model).
7. moving reused permissionDeniedCell and its model to more generic place in the project folder, therefore it can now be reused.

## Link to Jira
Main Story:
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-7648
Subtask:
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-7812

## Screenshots
Empty State:
7812-LM-4
7812-DM-4

Missing permissions:
7812-LM-3
7812-DM-3

State of requestCertificate + healthCertifiedPersionCell (old one, to be redesigned in separate ticket)
7812-LM-1
7812-LM-2
7812-DM-1
7812-DM-2

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Views/EmptyStateView/EmptyStateView.swift [hunk 1]
```diff
@@ -83,7 +83,7 @@ class EmptyStateView: UIView {
 		stackView.addArrangedSubview(descriptionLabel)
 
 		topConstraint = containerView.topAnchor.constraint(greaterThanOrEqualTo: topAnchor)
-
+		let percentageWidth = UIScreen.main.bounds.width / 3
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/Views/EmptyStateView/EmptyStateView.swift:86]: where does this magic number 3 come from? Maybe add a comment.

**Review by KaiTeuber — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Views/EmptyStateView/EmptyStateView.swift (hunk 1 above)
Comment (pascal-brause) [reply src/xcode/ENA/ENA/Source/Views/EmptyStateView/EmptyStateView.swift:86]: I searched for a divider that fits for all four occurrences for the EmptyStateView (Certificates, Checkins, Contact Diary, Trace Locations).
But yes, good point, i will leave a comment.

File: src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings [hunk 2]
```diff
@@ -2590,13 +2590,13 @@ Die Angaben werden statistisch ausgewertet, sie werden nicht zu einem Profil ges
...
+"HealthCertificate_Overview_add" = "Zertifikat hinzufügen";
 
-/* - Vaccination Certificate Registration */
+"HealthCertificate_Overview_EmptyTitle" = "Noch keine Zertifikate vorhanden";
 
-"VaccinationCertificate_Registration_title" = "EU Digitalen\n**COVID-Impfnachweis hinzufügen**";
+"HealthCertificate_Overview_EmptyDescription" = "Hier werden die gespeicherten digitalen Impfzertifikate, Genesenenzertifikate und Testzertifikate angezeigt.";
 
-"VaccinationCertificate_Registration_description" = "Fügen Sie Ihre Impfzertifikate in die App hinzu, um sie immer bei sich zu haben. Scannen Sie dafür den QR-Code Ihres Dokuments.";
+"HealthCertificate_Overview_EmptyImageDescription" = "Zertifikatssigel mit digitalem Zertifikat im Hintergrund";
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings:2599]:
```suggestion
"HealthCertificate_Overview_EmptyImageDescription" = "Zertifikatssiegel mit digitalem Zertifikat im Hintergrund";
```

File: src/xcode/ENA/ENA/Source/Scenes/Events/Cells/AddEventTableViewCell.swift [hunk 3]
```diff
@@ -35,7 +35,7 @@ class AddEventTableViewCell: UITableViewCell {
 
 	// MARK: - Internal
 
-	func configure(cellModel: AddEventCellModel) {
+	func configure(cellModel: AddCellModeling) {
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/Events/Cells/AddEventTableViewCell.swift:38]: Would it make sense to rename this cell to `AddTableViewCell` then?

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/CreateCertificate/AddCertificateTableViewCell.swift [hunk 4]
```diff
@@ -0,0 +1,67 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import UIKit
+import OpenCombine
+
+class AddCertificateTableViewCell: UITableViewCell, ReuseIdentifierProviding {
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/CreateCertificate/AddCertificateTableViewCell.swift:8]: Do we need a copy or could we also use the existing cell?

**Review by nickguendling — CHANGES_REQUESTED** [review]

**Review by NataliaLemmerth — APPROVED** [review]
UA approved

File: src/xcode/ENA/ENA/Source/Scenes/Events/Cells/AddEventTableViewCell.swift (hunk 3 above)
Comment (pascal-brause) [reply src/xcode/ENA/ENA/Source/Scenes/Events/Cells/AddEventTableViewCell.swift:38]: Yes, we can :)

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/CreateCertificate/AddCertificateTableViewCell.swift (hunk 4 above)
Comment (pascal-brause) [reply src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/CreateCertificate/AddCertificateTableViewCell.swift:8]: Good idea, I will have a look if I can refactor that, too.

**Review by naveeddotio — CHANGES_REQUESTED** [review]
UI Tests are failing after the refactoring.

**Review by nickguendling — APPROVED** [review]

**Review by naveeddotio — APPROVED** [review]

**Pull request merged**